woocommerce/woocommerce-ios
# `ProductVariation` now conforms to `ProductFormDataModel` and `TaxClassRequestable` for price settings customization

Prerequisite for #2085 

## Changes

- Updated `ProductVariation` to conform to `ProductFormDataModel` and `TaxClassRequestable` so that it can be passed to `ProductPriceSettingsViewModel`
- In `ProductPriceSettingsViewModel`, updated its sections logic so that `Product` form includes a tax section while `ProductVariation` form does not
  - I didn't really like how it checks the concrete type of `product` to differentiate the types, please let me know if you have other ideas!
- Added `MockProductVariation` to `WooCommerceTests` target for tests
- Created `ProductPriceSettingsViewModel_ProductVariationTests` that tests on the sections logic (no tax section)
- Created `ProductVariation_ProductFormTests` that tests on the derived properties of `ProductVariation`

## Testing

- Go to the Products tab
- Tap on a simple product
- Tap on the price settings row, potentially edit some fields --> everything should work as before

Update release notes:
